Use a better insertion point for buffer descriptor loads

jayfoad opened the pull request:
When inserting buffer descriptor loads and push constant loads, insert them at the nearest common dominator of all uses, instead of always in the entry block.

Currently we run an IR code sinking pass later which would have sunk these loads anyway, but we would like to stop running that pass because it can have bad effects on register pressure.

nhaehnle commented:
Looks reasonable, except `findNearestCommonDominator` seems to be missing?

jayfoad commented:
> Looks reasonable, except `findNearestCommonDominator` seems to be missing?

Ugh, must be merge damage from a late cherry-pick onto gpuopen/dev. Will fix.

amdvlk-admin commented:
### Test summary for commit 8d3d094a950c0695fc0fbd40280e457839af1955
<details><summary>CTS tests (Failed: 350/207253)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38568/70466 (54.7%)</li>
<li><details><summary>Failed: 348/70466 (0.5%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.uniform_texel.storage_texel.uniform.storage.index.vertex.indirect.create.binding_residency
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r32_sfloat.r16g16_sfloat.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r8g8b8a8_unorm.a2r10g10b10_uint_pack32.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_buffer_with_offset.6_5
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.array.array_to_array_layers
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.dimensions.src8192x4_dst8192x4.r4g4_unorm_pack8.r4g4_unorm_pack8.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.multiple_layers.a8b8g8r8_snorm_pack32_multiple_subresourcerange
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.2d.optimal.remaining_array_layers.b5g5r5a1_unorm_pack16_64x11_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.buffer_device_address.set3.depth1.baseubo.crossconvertu2p.store.replay.scalar.vert
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptor_copy.compute.uniform_buffer_3
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptor_update.random.uniform_buffer
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets16.noarray.ubolimithigh.sbolimitlow.sampledimglow.outimgtexlow.iublimitlow.nouab.rgen.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets8.unifindexed.ubolimitlow.sbolimitlow.sampledimghigh.lowimgnotex.noiub.uab.call.noia.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_immutable.vertex.multiple_descriptor_sets.single_descriptor.1d_array_base_mip
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.alias.u32_array_to_f64_scalar
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.u8vec4_array_to_uint16_t_float16_t_array_1
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint16_t_array_to_uvec3_uint_array_2
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.draw.renderpass.basic_draw.draw_indexed.line_strip_with_adjacency.1
Stack trace: Fail
...
```
</p></details></li>

<li>Not Supported: 31550/70466 (44.8%)</li><li>Warnings: 0/70466 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35749/70474 (50.7%)</li>
<li><details><summary>Failed: 1/70474 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.spirv_assembly.instruction.compute.variable_pointers.complex_types_compute.opphi_outer_struct_two_buffers_second_input
Stack trace: Crash
```
</p></details></li>

<li>Not Supported: 34724/70474 (49.3%)</li><li>Warnings: 0/70474 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33475/66313 (50.5%)</li>
<li><details><summary>Failed: 1/66313 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.spirv_assembly.instruction.compute.variable_pointers.complex_types_compute.opphi_outer_struct_two_buffers_second_input
Stack trace: Crash
```
</p></details></li>

<li>Not Supported: 32837/66313 (49.5%)</li><li>Warnings: 0/66313 (0.0%)</li>
</ul></details>


</ul></details>

nhaehnle reviewed (approved):
LGTM

nhaehnle reviewed (dismissed):
Makes sense.

amdvlk-admin commented:
### Test summary for commit 62a8087223994ab22b5426105e0865e33a54e287
<details><summary>CTS tests (Failed: 331/207247)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38581/70464 (54.8%)</li>
<li><details><summary>Failed: 331/70464 (0.5%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.storage_texel.create.binding_residency
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.uniform_texel.storage_texel.uniform.storage.vertex.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r32_sfloat.r32_sfloat.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_buffer_with_offset.6_7
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.array.array_to_array_whole_mipmap_r8g8b8a8_uint
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.dimensions.src8192x4_dst8192x4.r4g4_unorm_pack8.r8_srgb.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.multiple_layers.a8b8g8r8_srgb_pack32_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.buffer_device_address.set3.depth3.baseubo.convert.nostore.replay.std140.comp
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptor_update.acceleration_structure.ray_query.regular.tesc
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets16.noarray.ubolimithigh.sbolimitlow.sampledimglow.outimgtexlow.iublimithigh.uab.vert.noia.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets8.unifindexed.ubolimitlow.sbolimitlow.sampledimghigh.lowimgsingletex.noiub.nouab.frag.noia.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_immutable.tess_ctrl.multiple_descriptor_sets.single_descriptor.1d_array
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance_dynamic_index.vert_tess_geom.7_1
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint16_t_array_to_uint8_t_vec4_array_2
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.5_4_1
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.7_5_2
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.descriptor_indexing.storage_buffer_dynamic
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31552/70464 (44.8%)</li><li>Warnings: 0/70464 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70472 (50.7%)</li>
<li>Failed: 0/70472 (0.0%)</li>
<li>Not Supported: 34722/70472 (49.3%)</li><li>Warnings: 0/70472 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66311 (50.5%)</li>
<li>Failed: 0/66311 (0.0%)</li>
<li>Not Supported: 32835/66311 (49.5%)</li><li>Warnings: 0/66311 (0.0%)</li>
</ul></details>


</ul></details>

jayfoad commented:
retest this please

dstutt commented:
retest this please

amdvlk-admin commented:
### Test summary for commit 639869ab1c7dbb0065c20c917b19ab638dea7bd3
<details><summary>CTS tests (Failed: 400/207247)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38513/70464 (54.7%)</li>
<li><details><summary>Failed: 400/70464 (0.6%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.storage.index.vertex.create.binding_residency_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.storage_texel.create.binding_residency
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.uniform_texel.storage_texel.uniform.storage.vertex.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.storage_texel.uniform.storage.index.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r32_sfloat.r32_sfloat.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_buffer_with_offset.6_7
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.a2b10g10r10_unorm_pack32.a2b10g10r10_unorm_pack32.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.dimensions.src6x16384_dst6x16384.r8_srgb.r4g4_unorm_pack8.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.array.array_to_array_whole_mipmap_r8g8b8a8_uint
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.dimensions.src8192x4_dst8192x4.r4g4_unorm_pack8.r8_srgb.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.linear.multiple_layers.a1r5g5b5_unorm_pack16
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.optimal.remaining_array_layers_twostep.b4g4r4a4_unorm_pack16_200x180_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.linear.single_layer.a2b10g10r10_uint_pack32_71x1
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.multiple_layers.a8b8g8r8_srgb_pack32_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.single_layer.r8_snorm_71x1_multiple_subresourcerange
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.2d.optimal.single_layer.r16g16b16a16_uint_64x11
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.3d.optimal.single_layer.b8g8r8a8_uint
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_depth_stencil_attachment.single_layer.d16_unorm_s8_uint_separate_layouts_depth_33x128
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.buffer_device_address.set3.depth3.baseubo.convert.nostore.replay.std140.comp
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.buffer_device_address.set31.depth1.baseubo.crossconvertp2u.nostore.replay.std140.comp
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31551/70464 (44.8%)</li><li>Warnings: 0/70464 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70472 (50.7%)</li>
<li>Failed: 0/70472 (0.0%)</li>
<li>Not Supported: 34722/70472 (49.3%)</li><li>Warnings: 0/70472 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66311 (50.5%)</li>
<li>Failed: 0/66311 (0.0%)</li>
<li>Not Supported: 32835/66311 (49.5%)</li><li>Warnings: 0/66311 (0.0%)</li>
</ul></details>


</ul></details>

ruiling commented on llpc/lower/llpcSpirvLowerGlobal.cpp:
```diff
@@ -1931,8 +1947,20 @@ void SpirvLowerGlobal::lowerBufferBlock() {
             continue;
 
           usesToReplace.push_back(inst);
+          insertPoint = findNearestCommonDominator(insertPoint, inst, domTree);
```
I am not sure if the `inst` might sit in the newly created return block during `lowerOutput()`? If this is possible, I think we may need to update the dominator tree there.

jayfoad replied:
I don't know if this can happen. I haven't seen any CTS failures caused by it. (The CI failures are "Flake" not "Fail" and anyway I can't reproduce them on my own Navi21 machine.)

qiaojbao commented:
retest this please

amdvlk-admin commented:
### Test summary for commit 639869ab1c7dbb0065c20c917b19ab638dea7bd3
<details><summary>CTS tests (Failed: 403/207247)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38513/70464 (54.7%)</li>
<li><details><summary>Failed: 403/70464 (0.6%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.storage.index.vertex.create.binding_residency_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.storage_texel.create.binding_residency
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.uniform_texel.storage.index.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_dst.uniform_texel.storage_texel.uniform.storage.vertex.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.storage_texel.uniform.storage.index.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.storage_texel.uniform.storage.index.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.uniform.vertex.create.binding
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.color.2d.r4g4_unorm_pack8.b8g8r8a8_snorm.optimal_linear_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r32_sfloat.r32_sfloat.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.simple_tests.mirror_subregions_3d.r32_sfloat_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_buffer_with_offset.6_7
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.b5g6r5_unorm_pack16.r16_snorm.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.dimensions.src6x16384_dst6x16384.r8_srgb.r4g4_unorm_pack8.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.2d.r4g4_unorm_pack8.r16g16_snorm.optimal_linear_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.array.array_to_array_whole_mipmap_r8g8b8a8_uint
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.dimensions.src4x8192_dst4x8192.r8g8b8a8_unorm.r32_sfloat.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.image_to_image.dimensions.src8192x4_dst8192x4.r4g4_unorm_pack8.r8_srgb.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.linear.multiple_layers.a1r5g5b5_unorm_pack16
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.optimal.remaining_array_layers_twostep.b4g4r4a4_unorm_pack16_200x180_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.linear.single_layer.a2b10g10r10_uint_pack32_71x1
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31548/70464 (44.8%)</li><li>Warnings: 0/70464 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70472 (50.7%)</li>
<li>Failed: 0/70472 (0.0%)</li>
<li>Not Supported: 34722/70472 (49.3%)</li><li>Warnings: 0/70472 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66311 (50.5%)</li>
<li>Failed: 0/66311 (0.0%)</li>
<li>Not Supported: 32835/66311 (49.5%)</li><li>Warnings: 0/66311 (0.0%)</li>
</ul></details>


</ul></details>

qiaojbao commented:
This PR seems cause navi21 test agent hang.

qiaojbao commented:
retest this please

qiaojbao commented:
retest this please

qiaojbao commented:
[70246/70246] Pass: 38460 Fail: 322 Skip: 20130 ExpectedFail: 0 UnexpectedPass: 0 Crash: 0 Timeout: 0 Missing: 11200 Flake: 134 Duration: 11:58 Remaining: 0
Still see 322 failures with navi21

amdvlk-admin commented:
### Test summary for commit 9ddccb48ec3f25177ed1be6ffd236ed813956dd2
<details><summary>CTS tests (Failed: 456/206581)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38460/70246 (54.8%)</li>
<li><details><summary>Failed: 456/70246 (0.6%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.uniform_texel.index.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.uniform_texel.storage_texel.vertex.indirect.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.buffer.suballocation.uniform_texel.create.binding
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.color.2d.e5b9g9r9_ufloat_pack32.a2b10g10r10_unorm_pack32.optimal_optimal_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.color.3d.a8b8g8r8_srgb_pack32.a8b8g8r8_srgb_pack32.general_general_linear_stripes_y
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_1.r32g32b32a32_uint.general_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_6.r8g8_snorm.optimal_optimal_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.all_formats.color.2d.r4g4b4a4_unorm_pack16.r5g5b5a1_unorm_pack16.optimal_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.dimensions.src4096x4_dst4096x4.r4g4_unorm_pack8.r8_srgb.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.resolve_image.whole.2_bit
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.blit_image.all_formats.color.2d.bc7_unorm_block.r5g6b5_unorm_pack16.optimal_general_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.buffer_to_buffer_with_offset.2_6
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.r4g4b4a4_unorm_pack16.r16_uint.general_optimal
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.core.image_to_image.all_formats.color.2d.r8g8b8a8_uint.r16g16_sfloat.general_general
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.color.1d.r16_snorm.r16_snorm.general_general_nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.simple_tests.mirror_subregions.r32_sfloat_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.simple_tests.mirror_xy_3d.nearest
Stack trace: Flake

FAILURE: dEQP-VK.api.fill_and_update_buffer.suballocation.fill_buffer_vk_whole_size_1_extra_bytes_offset_4
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.1d.linear.single_layer.b8g8r8a8_srgb
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.partial_clear_color_attachment.single_layer.r5g6b5_unorm_pack16_200x180_clamp_input
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31330/70246 (44.6%)</li><li>Warnings: 0/70246 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70248 (50.9%)</li>
<li>Failed: 0/70248 (0.0%)</li>
<li>Not Supported: 34498/70248 (49.1%)</li><li>Warnings: 0/70248 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66087 (50.7%)</li>
<li>Failed: 0/66087 (0.0%)</li>
<li>Not Supported: 32611/66087 (49.3%)</li><li>Warnings: 0/66087 (0.0%)</li>
</ul></details>


</ul></details>

jayfoad commented:
retest this please

qiaojbao commented:
retest this please

qiaojbao commented:
retest this please

amdvlk-admin commented:
### Test summary for commit 7ef2d36018ede1e50c2f17cb0fffe57dbbd08ec1
<details><summary>CTS tests (Failed: 319/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38595/70473 (54.8%)</li>
<li><details><summary>Failed: 319/70473 (0.5%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.uniform_texel.index.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_6.a8b8g8r8_unorm_pack32.general_optimal_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.dimensions.src4096x4_dst4096x4.r4g4b4a4_unorm_pack16.r16_sfloat.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.optimal.remaining_array_layers.a4r4g4b4_unorm_pack16_33x128
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.linear.single_layer.b4g4r4a4_unorm_pack16_200x1_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.partial_clear_color_attachment.single_layer.b4g4r4a4_unorm_pack16_64x11_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.noarray.ubolimitlow.sbolimitlow.sampledimglow.outimgtexlow.iublimitlow.uab.frag.noia.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.runtimesize.ubolimitlow.sbolimitlow.sampledimglow.lowimgnotex.iublimithigh.uab.chit.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets8.unifindexed.ubolimitlow.nosbo.sampledimghigh.outimgtexlow.noiub.uab.frag.ialimitlow.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_mutable.no_access.multiple_descriptor_sets.multiple_contiguous_descriptors.cube_array_base_slice
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Fail

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint_array_to_double_vec4_array_3
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_5
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.geometry.emit.triangle_strip_emit_1_end_1
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.1d_array.12_1_6.render_different_content
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.2d_array.64_64_4.multiple_layers_per_invocation
Stack trace: Fail

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_in.ivec4.as_int_int_int_int
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31559/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

qiaojbao commented:
retest this please

qiaojbao commented:
Maybe these failures caused by the branch too old.
Could you try to rebase it to the latest?

jayfoad commented:
> Maybe these failures caused by the branch too old. Could you try to rebase it to the latest?

It was not old; I rebased it less than 24 hours ago! Anyway I have rebased it again.

amdvlk-admin commented:
### Test summary for commit 68d7161f68fb1c02f31d38d9ff6b42303e0cbc6f
<details><summary>CTS tests (Failed: 294/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38619/70473 (54.8%)</li>
<li><details><summary>Failed: 294/70473 (0.4%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer.suballocation.transfer_src.transfer_dst.uniform_texel.index.create.binding_aliased
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_6.a8b8g8r8_unorm_pack32.general_optimal_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.copy_and_blit.copy_commands2.image_to_image.dimensions.src4096x4_dst4096x4.r4g4b4a4_unorm_pack16.r16_sfloat.optimal_general
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.2d.optimal.remaining_array_layers.a4r4g4b4_unorm_pack16_33x128
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.linear.single_layer.b4g4r4a4_unorm_pack16_200x1_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.partial_clear_color_attachment.single_layer.b4g4r4a4_unorm_pack16_64x11_clamp_input
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.runtimesize.ubolimitlow.sbolimitlow.sampledimglow.lowimgnotex.iublimithigh.uab.chit.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_mutable.no_access.multiple_descriptor_sets.multiple_contiguous_descriptors.cube_array_base_slice
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Fail

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint_array_to_double_vec4_array_3
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_5
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.geometry.emit.triangle_strip_emit_1_end_1
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.1d_array.12_1_6.render_different_content
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.2d_array.64_64_4.multiple_layers_per_invocation
Stack trace: Fail

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_in.ivec4.as_int_int_int_int
Stack trace: Flake

FAILURE: dEQP-VK.glsl.arrays.unnamed_parameter.float_mat3_vertex
Stack trace: Fail

FAILURE: dEQP-VK.glsl.atomic_operations.add_unsigned_geometry_reference
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31560/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
### Test summary for commit 68d7161f68fb1c02f31d38d9ff6b42303e0cbc6f
<details><summary>CTS tests (Failed: 314/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38599/70473 (54.8%)</li>
<li><details><summary>Failed: 314/70473 (0.4%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_1.r5g6b5_unorm_pack16.general_general_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.1d.linear.multiple_layers.b5g5r5a1_unorm_pack16_71x1
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.single_layer.r16_uint_71x1
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.noarray.ubolimitlow.sbolimitlow.sampledimglow.outimgtexlow.iublimitlow.uab.frag.noia.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.runtimesize.ubolimitlow.sbolimitlow.sampledimglow.lowimgnotex.iublimithigh.uab.chit.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets8.unifindexed.ubolimitlow.nosbo.sampledimghigh.outimgtexlow.noiub.uab.frag.ialimitlow.0
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_mutable.no_access.multiple_descriptor_sets.multiple_contiguous_descriptors.cube_array_base_slice
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Fail

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.alias.u32_array_to_f64_scalar
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint_array_to_double_vec4_array_3
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_5
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.draw.renderpass.instanced.draw_indexed_vk_primitive_topology_triangle_list_attrib_divisor_4_multiview
Stack trace: Flake

FAILURE: dEQP-VK.geometry.emit.triangle_strip_emit_1_end_1
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.1d_array.12_1_6.render_different_content
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.2d_array.64_64_4.multiple_layers_per_invocation
Stack trace: Fail

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_in.ivec4.as_int_int_int_int
Stack trace: Flake

FAILURE: dEQP-VK.glsl.arrays.unnamed_parameter.float_mat3_vertex
Stack trace: Fail
...
```
</p></details></li>

<li>Not Supported: 31560/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
### Test summary for commit 326fa30bb98435cdd9d785b777119fc18ba8c181
<details><summary>CTS tests (Failed: 283/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38631/70473 (54.8%)</li>
<li><details><summary>Failed: 283/70473 (0.4%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.runtimesize.ubolimitlow.sbolimitlow.sampledimglow.lowimgnotex.iublimithigh.uab.chit.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_mutable.no_access.multiple_descriptor_sets.multiple_contiguous_descriptors.cube_array_base_slice
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Fail

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint_array_to_double_vec4_array_3
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_5
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.geometry.emit.triangle_strip_emit_1_end_1
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.1d_array.12_1_6.render_different_content
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.2d_array.64_64_4.multiple_layers_per_invocation
Stack trace: Fail

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_in.ivec4.as_int_int_int_int
Stack trace: Flake

FAILURE: dEQP-VK.glsl.arrays.unnamed_parameter.float_mat3_vertex
Stack trace: Fail

FAILURE: dEQP-VK.glsl.atomic_operations.add_unsigned_geometry_reference
Stack trace: Flake

FAILURE: dEQP-VK.glsl.atomic_operations.exchange_signed_vertex_reference
Stack trace: Flake

FAILURE: dEQP-VK.glsl.builtin.precision.acos.mediump.scalar
Stack trace: Flake

FAILURE: dEQP-VK.glsl.constant_expressions.complex_types.nested_struct_vertex
Stack trace: Fail

FAILURE: dEQP-VK.glsl.constant_expressions.operators.compare_vertex
Stack trace: Flake

FAILURE: dEQP-VK.glsl.loops.special.while_uniform_iterations.vector_counter_vertex
Stack trace: Flake

FAILURE: dEQP-VK.glsl.matrix.div.const.highp_mat3_float_fragment
Stack trace: Fail
...
```
</p></details></li>

<li>Not Supported: 31559/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

ruiling reviewed (commented):
LGTM, Thanks!

amdrexu commented on llpc/lower/llpcSpirvLowerGlobal.cpp:
```diff
@@ -1861,6 +1870,30 @@ void SpirvLowerGlobal::storeOutputMember(Type *outputTy, Value *storeValue, Arra
   llvm_unreachable("Should never be called!");
 }
 
+// Find a new insertion point that dominates the current insertion point \p instA and a specified instruction use
```
A nit: Could you add the missing separate line please?

amdrexu commented on llpc/lower/llpcSpirvLowerGlobal.cpp:
```diff
@@ -211,18 +215,23 @@ bool LegacySpirvLowerGlobal::runOnModule(Module &module) {
 // =====================================================================================================================
 // Executes this SPIR-V lowering pass on the specified LLVM module.
 //
 // @param [in/out] module : LLVM module to be run on
```
Please update the parameter comments. A new parameter is added.

amdrexu commented on llpc/lower/llpcSpirvLowerGlobal.cpp:
```diff
@@ -1861,6 +1870,30 @@ void SpirvLowerGlobal::storeOutputMember(Type *outputTy, Value *storeValue, Arra
+    return instA->comesBefore(instB) ? instA : instB;
+
+  BasicBlock *block = domTree.findNearestCommonDominator(blockA, blockB);
+  return block == blockA ? instA : block == blockB ? instB : block->getTerminator();
```
Better to add a () for clearness
->
(block == blockB ? instB : block->getTerminator())

nhaehnle commented:
Do the tests pass now?

jayfoad commented:
> Do the tests pass now?

No, there seem to be some genuine failures caused by latent problems with phi handling in PatchBufferOp. I'll put up a separate patch for that soon...

amdvlk-admin commented:
### Test summary for commit 3eb5f7b4e9a5bcd86ed58432eac345274bf2b8ba
<details><summary>CTS tests (Failed: 282/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38631/70473 (54.8%)</li>
<li><details><summary>Failed: 282/70473 (0.4%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.copy_and_blit.dedicated_allocation.blit_image.all_formats.generate_mipmaps.from_base_level.layercount_1.r5g6b5_unorm_pack16.general_general_linear
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.core.clear_color_image.1d.linear.multiple_layers.b5g5r5a1_unorm_pack16_71x1
Stack trace: Flake

FAILURE: dEQP-VK.api.image_clearing.dedicated_allocation.clear_color_image.1d.optimal.single_layer.r16_uint_71x1
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.descriptorset_random.sets4.runtimesize.ubolimitlow.sbolimitlow.sampledimglow.lowimgnotex.iublimithigh.uab.chit.noia.0
Stack trace: Fail

FAILURE: dEQP-VK.binding_model.shader_access.secondary_cmd_buf.sampler_mutable.no_access.multiple_descriptor_sets.multiple_contiguous_descriptors.cube_array_base_slice
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_cull_distance.vert_geom.7_1_fragmentshader_read
Stack trace: Fail

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uint_array_to_double_vec4_array_3
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_5
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.3_6_6
Stack trace: Fail

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Flake

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat3x3
Stack trace: Flake

FAILURE: dEQP-VK.geometry.emit.triangle_strip_emit_1_end_1
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.1d_array.12_1_6.render_different_content
Stack trace: Fail

FAILURE: dEQP-VK.geometry.layered.2d_array.64_64_4.multiple_layers_per_invocation
Stack trace: Fail

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_in.ivec4.as_int_int_int_int
Stack trace: Flake

FAILURE: dEQP-VK.glsl.arrays.unnamed_parameter.float_mat3_vertex
Stack trace: Fail

FAILURE: dEQP-VK.glsl.atomic_operations.add_unsigned_geometry_reference
Stack trace: Fail

FAILURE: dEQP-VK.glsl.atomic_operations.exchange_signed_vertex_reference
Stack trace: Flake

FAILURE: dEQP-VK.glsl.builtin.precision.acos.mediump.scalar
Stack trace: Flake

FAILURE: dEQP-VK.glsl.builtin_var.fragdepth.line_list_d32_sfloat_multisample_8
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31560/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

jayfoad commented on llpc/lower/llpcSpirvLowerGlobal.h:
```diff
@@ -118,6 +123,8 @@ class SpirvLowerGlobal : public SpirvLower, public llvm::PassInfoMixin<SpirvLowe
   void interpolateInputElement(unsigned interpLoc, llvm::Value *interpInfo, llvm::CallInst &callInst,
                                GlobalVariable *gv, ArrayRef<Value *> indexOperands);
 
+  function_ref<DominatorTree &(Function &)> m_lookupDomTree;
```
Now that we have dropped the old pass manager there is no need for this lookupDomTree abstraction. I will clean this up.

nhaehnle reviewed (approved):
LGTM
